lib/src/Uncertainty/Distribution/PointConditionalDistribution.cxx

```diff
@@ -0,0 +1,648 @@
+Point PointConditionalDistribution::getStandardDeviation() const
+{
+  if (useSimplifiedVersion_)
+    return simplifiedVersion_.getStandardDeviation();
+
+  if (distribution_.hasIndependentCopula())
+    return marginalConditionedDistribution_.getStandardDeviation();
+  else
+    return DistributionImplementation::getStandardDeviation();
+}
+
+/* Get the skewness of the distribution */
+Point PointConditionalDistribution::getSkewness() const
+{
+  if (useSimplifiedVersion_)
+    return simplifiedVersion_.getSkewness();
+
+  if (distribution_.hasIndependentCopula())
+    return marginalConditionedDistribution_.getSkewness();
+  else
+    return DistributionImplementation::getSkewness();
+}
+
+/* Get the kurtosis of the distribution */
+Point PointConditionalDistribution::getKurtosis() const
+{
+  if (useSimplifiedVersion_)
+    return simplifiedVersion_.getKurtosis();
+
+  if (distribution_.hasIndependentCopula())
+    return marginalConditionedDistribution_.getKurtosis();
+  else
+    return DistributionImplementation::getKurtosis();
+}
+
+/* Compute the covariance of the distribution */
+void PointConditionalDistribution::computeCovariance() const
+{
+  if (useSimplifiedVersion_)
+  {
+    covariance_ = simplifiedVersion_.getCovariance();
+    isAlreadyComputedCovariance_ = true;
+    return;
+  }
+
+  if (distribution_.hasIndependentCopula())
+  {
+    covariance_ = marginalConditionedDistribution_.getCovariance();
+    isAlreadyComputedCovariance_ = true;
+  }
+  else
+    DistributionImplementation::computeCovariance();
+}
+
+
+/* Get the i-th marginal distribution */
+Distribution PointConditionalDistribution::getMarginal(const UnsignedInteger index) const
+{
+  if (useSimplifiedVersion_)
+    return simplifiedVersion_.getMarginal(index);
+
+  const UnsignedInteger dimension = getDimension();
+  if (index >= dimension) throw InvalidArgumentException(HERE) << "The index of a marginal distribution must be in the range [0, dim-1]";
+  if (dimension == 1) return clone();
+
+  if (distribution_.hasIndependentCopula())
+    return marginalConditionedDistribution_.getMarginal(index);
+  else
+    return DistributionImplementation::getMarginal(index);
+}
+
+/* Get the distribution of the marginal distribution corresponding to indices dimensions */
+Distribution PointConditionalDistribution::getMarginal(const Indices & indices) const
+{
+  if (useSimplifiedVersion_)
+    return simplifiedVersion_.getMarginal(indices);
+
+  const UnsignedInteger dimension = getDimension();
+  if (!indices.check(dimension)) throw InvalidArgumentException(HERE) << "The indices of a marginal distribution must be in the range [0, dim-1] and must be different";
+
+  Indices full(dimension);
+  full.fill();
+  if (indices == full) return clone();
+
+  if (distribution_.hasIndependentCopula())
+    return marginalConditionedDistribution_.getMarginal(indices);
+  else
+    return DistributionImplementation::getMarginal(indices);
```

Comment:
The marginal distribution is a PointConditionalDistribution with the same conditioning indices and values, but on a subset of the non conditioned indices